### DISCUSSION TITLE
Translation of button names

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#157] [Ideas]
@Laravel-Lang/laravel-lang, hey!

I had an idea to create another project in which we can add names for interface buttons.

In this regard, I had a question: what is better to name the project - `laravel-lang/actions` or `laravel-lang/buttons`?

To understand the context, I found a screenshot on the Internet with an example of such buttons. Of course, there will be more of them in the project and those that can be used in real projects.

![image](https://github.com/Laravel-Lang/common/assets/10347617/ae46e46e-28c4-42a1-a4d5-e6c61187ec0d)

When installing a dependency, you can save the values both in a JSON file and in PHP:

```json
{
    "Detail": "Detail",
    "Edit": "Edit",
    "Export": "Export",
    "New": "New"
}
```
```php
return [
    'detail' => 'Detail',
    'edit'   => 'Edit',
    'export' => 'Export',
    'new'    => 'New',
];
```

What do you think about this idea?

**Comment by mohamedsabil83** [comment]
Good Idea. I prefer **laravel-lang/actions**; we can add a lot with this name, unlike **laravel-lang/buttons**.

**Comment by Oleksandr-Moik** [comment]
Yeah, 'actions' name is more general. I think it's commonly used for admin panels but also can be useful in other contexts in projects.

**Comment by andrey-helldar** [comment]
From my practice, I suggest starting with such an array of names, and also doing them not only in PHP format, but also in JSON for ease of use:

```php
return [
    'accept'        => 'Accept',
    'action'        => 'Action',
    'actions'       => 'Actions',
    'add'           => 'Add',
    'admin'         => 'Admin',
    'agree'         => 'Agree',
    'archive'       => 'Archive',
    'assign'        => 'Assign',
    'attach'        => 'Attach',
    'browse'        => 'Browse',
    'cancel'        => 'Cancel',
    'choose'        => 'Choose',
    'choose_file'   => 'Choose File',
    'choose_image'  => 'Choose Image',
    'close'         => 'Close',
    'confirm'       => 'Confirm',
    'connect'       => 'Connect',
    'create'        => 'Create',
    'delete'        => 'Delete',
    'detach'        => 'Detach',
    'details'       => 'Details',
    'disable'       => 'Disable',
    'discard'       => 'Discard',
    'done'          => 'Done',
    'down'          => 'Down',
    'duplicate'     => 'Duplicate',
    'edit'          => 'Edit',
    'enable'        => 'Enable',
    'export'        => 'Export',
    'file'          => 'File',
    'files'         => 'Files',
    'go_home'       => 'Go Home',
    'hide'          => 'Hide',
    'home'          => 'Home',
    'image'         => 'Image',
    'Impersonate'   => 'Impersonate',
    'Impersonation' => 'Impersonation',
    'import'        => 'Import',
    'load'          => 'Load',
    'localize'      => 'Localize',
    'new'           => 'New',
    'no'            => 'No',
    'open'          => 'Open',
    'open_website'  => 'Open on the website',
    'preview'       => 'Preview',
    'restore'       => 'Restore',
    'save'          => 'Save',
    'search'        => 'Search',
    'select'        => 'Select',
    'select_all'    => 'Select All',
    'settings'      => 'Settings',
    'show'          => 'Show',
    'solve'         => 'Solve',
    'translate_it'  => 'Translate it',
    'unpack'        => 'Unpack',
    'up'            => 'Up',
    'update'        => 'Update',
    'user'          => 'User',
    'view'          => 'View',
    'yes'           => 'Yes',
    'named'         => [
        'choose'    => 'Choose :name',
        'edit'      => 'Edit :name',
        'hide'      => 'Hide :name',
        'import'    => 'Import :name',
        'new'       => 'New :name',
        'restore'   => 'Restore :name',
        'save'      => 'Save :name',
        'show'      => 'Show :name',
        'update'    => 'Update :name',
        'view'      => 'View :name',
        'duplicate' => 'Duplicate: name',
        'search'    => 'Search :name',
    ],
];
```

I did not post it in JSON form, because... the keys will be the same.

**Reply by andrey-helldar** [reply]
So far I have created a frame. I'll be making some initial changes soon.

https://github.com/Laravel-Lang/actions

I made the repository private so as not to scare people 😂

**Reply by andrey-helldar** [reply]
@Laravel-Lang/laravel-lang, hi!

I made the project public and configured it. Also launched machine translation for initial filling.

https://github.com/Laravel-Lang/actions

You can already check translations and add new keys. If you think that some key should be in the package, feel free to send it 🙂

```bash
composer require laravel-lang/actions
```

Tomorrow I plan to release release 1.0.0.

UPD: Translation status added to documentation: https://laravel-lang.com/statuses-actions.html

**Comment by es-ka** [comment]
I just stumbled upon this package by accident.
It seems most German translations are done by machines with no humans involved and some are wrong in the context being used with buttons: e.g. `"Load": "Belastung"`

I'd really like to improve the German ones, but I have some questions.
The description of the package is really short and not quite clear to me:
> The package includes translation of buttons and interface elements.

Is this package being used by Laravel itself, any Laravel package or any 3rd-party package?
For some strings the context is important and I want to make sure not to break any Laravel app out there...

**Reply by andrey-helldar** [reply]
No, translations from this package are not used by Laravel products.

Any developer can propose their own names to be used in the application interface.

Regarding context, yes, there are places where context is important. Initially the "actions" package has translations for buttons like "show", "delete", "change". If some word doesn't fit the context, it's better to add a new key with a new translation.

And yes, the main part is machine translation from Google Translate. Language speakers try to keep translations up to date, but it's not always fast.

**Reply by sotten** [reply]
I think there is a misunderstanding here. The translation is already used by Laravel products.  For example, if you install Laravel with a Laravel frontend like Jetstream, then you can use this package to translate Jetstream.
I think it is correct to correct incorrect translations without a new key.

**Reply by andrey-helldar** [reply]
The quote is on the "Actions" package description page. It was conceived as a complement.

https://laravel-lang.com/packages-actions.html

But I agree that it is not recommended to fix existing keys.
